gordo_components/model/anomaly/diff.py

```diff
@@ -126,17 +134,49 @@ def cross_validate(
             )
             y_true = y.iloc[test_idxs] if isinstance(y, pd.DataFrame) else y[test_idxs]
 
-            diff = self._fold_thresholds(y_true=y_true, y_pred=y_pred, fold=i)
+            # Model's timestep scaled mse
+            scaled_mse = self._scaled_mse_per_timestep(split_model, y_true, y_pred)
+            scaled_mse_per_timestep = pd.concat((scaled_mse_per_timestep, scaled_mse))
 
             # Accumulate the rolling mins of diffs into common df
-            thresholds = thresholds.append(diff)
+            tag_thresholds_fold = self._feature_fold_thresholds(y_true, y_pred, fold=i)
+            feature_thresholds = feature_thresholds.append(tag_thresholds_fold)
 
-        # And finally, get the mean of all the pre-calculated thresholds over the folds
-        self.thresholds_ = self._final_thresholds(thresholds=thresholds)
+        # Calculate the final thresholds per feature based on the previous fold calculations
+        self.feature_thresholds_ = self._final_thresholds(thresholds=feature_thresholds)
+
+        # For the aggregate, use the accumulated mse of scaled residuals per timestep
+        self.aggregate_threshold_ = scaled_mse_per_timestep.rolling(6).min().max()
```

Comment:
Maybe as a way of compartmentalizing this out you could do a context manager thing {SWIP-ey pseudo-code to follow), That way it might be easier to swap the method used here and have that reviewed/owned a little more externally.
```
Class AnomolyMonitor:
    feature_thresholds = list()
    aggregate_threshold = None
    def __enter__(self):
        return self
    def _final_thresholds():
        # do stuff
    def _scaled_mse_timestep():
        # do stuff
    def collect(true, pred, fold):
        self.feature_thresholds.append(_feature_fold_thresholds(true, pred, fold))
    def __exit__(self):
        # calc means
        # calc aggregate, etc.

class DiffBased
    def cv(self):
        with AnomolyMontior() as am:
            am.collect(true, pred, fold)
            self.aggregate = am.aggregate